test/framework/vsphere.go

```diff
@@ -713,34 +645,54 @@ func WithBottlerocketFromRelease(release *releasev1.EksARelease, kubeVersion any
 
 func (v *VSphere) WithBottleRocketForRelease(release *releasev1.EksARelease, kubeVersion anywherev1.KubernetesVersion) api.ClusterConfigFiller {
 	return api.VSphereToConfigFiller(
-		api.WithTemplateForAllMachines(v.templateForRelease(anywherev1.Bottlerocket, release, kubeVersion)),
+		api.WithTemplateForAllMachines(v.templateForDevRelease(anywherev1.Bottlerocket, kubeVersion)),
 	)
 }
 
 func optionToSetTemplateForRelease(osFamily anywherev1.OSFamily, release *releasev1.EksARelease, kubeVersion anywherev1.KubernetesVersion) VSphereOpt {
 	return func(v *VSphere) {
 		v.fillers = append(v.fillers,
-			api.WithTemplateForAllMachines(v.templateForRelease(osFamily, release, kubeVersion)),
+			api.WithTemplateForAllMachines(v.templateForDevRelease(osFamily, kubeVersion)),
 		)
 	}
 }
 
-func envVarForVSphereTemplate(osFamily, eksDName string) string {
+// EnvVarForTemplate looks for explicit configuration through an env var: "T_VSPHERE_TEMPLATE_{osFamily}_{eks-d version}"
+// eg: T_VSPHERE_TEMPLATE_REDHAT_KUBERNETES_1_23_EKS_22.
+func (v *VSphere) EnvVarForTemplate(osFamily, eksDName string) string {
 	return fmt.Sprintf("T_VSPHERE_TEMPLATE_%s_%s", strings.ToUpper(osFamily), strings.ToUpper(strings.ReplaceAll(eksDName, "-", "_")))
 }
 
-func defaultNameForVSphereTemplate(templatesFolder, osFamily, eksDName string) string {
-	return filepath.Join(templatesFolder, fmt.Sprintf("%s-%s", strings.ToLower(eksDName), strings.ToLower(osFamily)))
+// DefaultNameForTemplate looks for a template with the name path: "{folder}/{eks-d version}-{osFamily}"
+// eg: /SDDC-Datacenter/vm/Templates/kubernetes-1-23-eks-22-redhat.
+func (v *VSphere) DefaultNameForTemplate(osFamily, eksDName string) string {
+	folder := v.testsConfig.TemplatesFolder
+	if folder == "" {
+		v.t.Log("vSphere templates folder is not configured, can't continue template search.")
```

Comment:
```suggestion
		v.t.Log("vSphere templates folder is not configured.")
```
IMO this method shouldn't assume where is being called from